archaius-core/src/main/java/com/netflix/archaius/config/AbstractConfig.java

```diff
@@ -201,130 +201,130 @@ public Short getShort(String key) {
             return Short.parseShort(value);
         }
         catch (NumberFormatException e) {
-            return parseError(e);
+            return parseError(key, value, e);
         }
     }
 
     @Override
     public Short getShort(String key, Short defaultValue) {
-        String value = getString(key);
+        String value = getRawString(key);
         if (value == null) 
             return notFound(defaultValue);
         try {
             return Short.parseShort(value);
         }
         catch (NumberFormatException e) {
-            return parseError(e, defaultValue);
+            return parseError(key, value, e);
         }
     }
 
     @Override
     public BigInteger getBigInteger(String key) {
-        String value = getString(key);
+        String value = getRawString(key);
         if (value == null) 
             return notFound();
         try {
             return BigInteger.valueOf(Long.valueOf(value));
         }
         catch (NumberFormatException e) {
-            return parseError(e);
+            return parseError(key, value, e);
         }
     }
 
     @Override
     public BigInteger getBigInteger(String key, BigInteger defaultValue) {
-        String value = getString(key);
+        String value = getRawString(key);
         if (value == null) 
-            return notFound();
+            return notFound(defaultValue);
         try {
             return BigInteger.valueOf(Long.valueOf(value));
         }
         catch (NumberFormatException e) {
-            return parseError(e, defaultValue);
+            return parseError(key, value, e);
         }
     }
 
     @Override
     public BigDecimal getBigDecimal(String key) {
-        String value = getString(key);
+        String value = getRawString(key);
         if (value == null) 
             return notFound();
         try {
             return BigDecimal.valueOf(Long.valueOf(value));
```

Comment:
Same as with `getBigInteger`, but it seems really strange here to force a conversion to Long first.